oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -746,15 +746,14 @@ private static Query nodeName(Filter.PropertyRestriction pr) {
         throw new IllegalStateException("For nodeName queries only EQUALS and LIKE are supported " + pr);
     }
 
-    private static Query like(String name, String first) {
-        first = first.replace('%', WildcardQuery.WILDCARD_STRING);
+    private static Query like(String name, String first) {first = first.replace('%', WildcardQuery.WILDCARD_STRING);
         first = first.replace('_', WildcardQuery.WILDCARD_CHAR);
 
-        int indexOfWS = first.indexOf(WildcardQuery.WILDCARD_STRING);
-        int indexOfWC = first.indexOf(WildcardQuery.WILDCARD_CHAR);
-        int len = first.length();
+        // If the query ends in a wildcard string (*) and has no other wildcard characters, use a prefix match query
+        boolean hasSingleWildcardStringAtEnd = first.indexOf(WildcardQuery.WILDCARD_STRING) == first.length() - 1;
+        boolean doesNotContainWildcardChar = first.indexOf(WildcardQuery.WILDCARD_CHAR) == -1;
```

Comment:
I usually use `first.indexOf(WildcardQuery.WILDCARD_CHAR) < 0` but `== -1` might be better

Reply:
The escape character problem applies here as well.

Reply:
I have no strong opinion here. I chose comparing with `-1` because it follows more closely the specification of `indexOf`, which states that it returns -1 if it does not find the character, so I believe it is more clear. `indexOf` should never return values less than `-1`, that would be a very serious problem from which we would like not be able to recover.